### PR TITLE
Documenting manually pushing metadata.

### PR DESCRIPTION
**Pull request by mpastell**
I thought it would be useful to add documentation about publishing METADATA manually. This is based on
https://groups.google.com/d/msg/julia-users/yEhxkDYxzI8/pPiW5SevKAYJ .

### CODE REVIEW

**Comment by mpastell** [comment]
Thanks for the suggestions, I made the changes.

**Comment by jiahao** [comment]
Looks good, please squash the commits and then I think we are good to go.

**Comment by mpastell** [comment]
I tried to squash, but apparently that lead to more commits... Any suggestions?

**Comment by jakebolewski** [comment]
Use `git rebase`

**Comment by mpastell** [comment]
I did, but that doesn't seem to remove already pushed commits for Github.

**Comment by jakebolewski** [comment]
You need to force push to this branch.

**Comment by jiahao** [comment]
After cleaning up your commits locally, you'll need to use `git push -f` to force git to update the remote branch, overwriting what it already knows remotely.

**Comment by timholy** [comment]
Depending on which version of git you're using, a safer syntax for force pushing is
```
git push <remotename> +<branchname>
```

**Comment by mpastell** [comment]
Thanks for the advice! I managed to do it now.

**Comment by tkelman** [comment]
OSX failure on Travis is unrelated but worrying:
```
	From worker 3:	     * core
Assertion failed: (false), function jl_llvmtuple_eltype, file ./intrinsics.cpp, line 215.
signal (6): Abort trap: 6
__pthread_kill at /usr/lib/system/libsystem_kernel.dylib (unknown line)
Worker 3 terminated.
ERROR: ProcessExitedException()
 in wait at /private/tmp/julia/lib/julia/sys.dylib (repeats 2 times)
 in wait_full at /private/tmp/julia/lib/julia/sys.dylib
 in remotecall_fetch at multi.jl:686
 in remotecall_fetch at multi.jl:691
 in anonymous at task.jl:1614
while loading /private/tmp/julia/share/julia/test/runtests.jl, in expression starting on line 42
```

File: doc/manual/packages.rst [hunk 1]
```diff
@@ -438,6 +439,38 @@ This means that :func:`Pkg.add("FooBar") <Pkg.add>` won't work yet since it only
 Moreover, when they run :func:`Pkg.update`, they will get the latest version of ``FooBar`` that you've pushed to the repo.
 This is a good way to have people test out your packages as you work on them, before they're ready for an official release.
 
+.. _man-manual-publish:
+
+Publishing METADATA manually
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If :func:`Pkg.publish` fails you can follow there instructions to
```
Comment (tkelman) [inline doc/manual/packages.rst]: these

File: doc/manual/packages.rst [hunk 2]
```diff
@@ -438,6 +439,38 @@ This means that :func:`Pkg.add("FooBar") <Pkg.add>` won't work yet since it only
...
+manually publish your package.
+
+By "forking" the main METADATA repository, you can create a
+personal copy (of METADATA.jl) under your GitHub account. Once
+that copy exists, you can push your local changes to your copy
+(just like any other GitHub project).
+
+
+1. go to  `<http://github.com/JuliaLang/METADATA.jl>`_ and create your own
```
Comment (tkelman) [inline doc/manual/packages.rst]: You could even link to `https://github.com/JuliaLang/METADATA.jl/fork` directly?

**Comment by mpastell** [comment]
@tkelman thanks for the fixes, I added them. I can't believe I made so many mistakes on such a small change.

**Comment by tkelman** [comment]
It's quite alright, thanks for the contribution!

**Comment by tkelman** [comment]
backported in 93bfc92a70890679e468bf83dbbd2ff020813fad

**Pull request merged**